File: app/src/main/java/huce/fit/appreadstories/story/filter/StoryFilterImpl.java
package huce.fit.appreadstories.story.filter;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

import huce.fit.appreadstories.api.Api;
import huce.fit.appreadstories.model.Truyen;
import huce.fit.appreadstories.story.BaseListStoryImpl;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class StoryFilterImpl extends BaseListStoryImpl implements StoryFilterPresenter {

    private static final String TAG = "StoryFilterImpl";
    private final StoryFilterView mStoryFilterView;
    private final List<Truyen> mListStory = new ArrayList<>();
    private int mSpeciesId, mStatusId;
    private final int mAge;
    private String mSpecies, mStatus;

    StoryFilterImpl(StoryFilterView storyFilterView, Context context) {
        super(context);
        mStoryFilterView = storyFilterView;
        mAge = getSharedPreferences().getAge();
    }

    @Override
    public void setSpeciesId(int mSpeciesId) {
        this.mSpeciesId = mSpeciesId;
    }

    @Override
    public void setStatusId(int mStatusId) {
        this.mStatusId = mStatusId;
    }

    @Override
    public int getSpeciesId() {
        return mSpeciesId;
    }

    @Override
    public int getStatusId() {
        return mStatusId;
    }

    @Override
    public void setSpecies(String species) {
        mSpecies = species;
    }

    @Override
    public void setStatus(String status) {
        mStatus = status;
    }

    @Override
    public void getData() {
        Log.e(TAG, "NHT mSpecies= " + mSpecies + " mStatus= " + mStatus);
        Api.apiInterface().getListStoriesFilter(mSpecies, mStatus, mAge).enqueue(new Callback<List<Truyen>>() {
            @Override
            public void onResponse(@NonNull Call<List<Truyen>> call, @NonNull Response<List<Truyen>> response) {
                if (response.isSuccessful() && response.body() != null) {
                    mListStory.clear();
                    mListStory.addAll(response.body());
                    mStoryFilterView.setData(response.body());
                }
            }

            @Override
            public void onFailure(@NonNull Call<List<Truyen>> call, Throwable t) {
                Log.e("Err_StoryFilter", "getDataFilter", t);
            }
        });
    }
}

File: app/src/main/java/huce/fit/appreadstories/story/list/filter/StoryFilterImpl.kt
package huce.fit.appreadstories.story.list.filter

import android.content.Context
import android.util.Log
import huce.fit.appreadstories.api.Api
import huce.fit.appreadstories.model.Story
import huce.fit.appreadstories.story.list.BaseStoryListImpl
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class StoryFilterImpl(val storyFilterView: StoryFilterView, context: Context) :
    BaseStoryListImpl(context), StoryFilterPresenter {

    companion object {
        private const val TAG = "StoryFilterImpl"
    }

    private var species = ""
    private var status = ""

    override fun setSpecies(species: String) {
        this.species = species
    }

    override fun setStatus(status: String) {
        this.status = status
    }

    override fun getData() {
        Log.e(TAG, "NHT mSpecies= $species status= $status")
        val age = getAccount().getAge()
        Api().apiInterface().getListStoriesFilter(species, status, age)
            .enqueue(object : Callback<List<Story>> {
                override fun onResponse(call: Call<List<Story>>, response: Response<List<Story>>) {
                    val storyServer = response.body()
                    if (response.isSuccessful && storyServer != null) {
                        storyFilterView.setData(storyServer)
                    }
                    Log.e(TAG, "api getData: success")
                }

                override fun onFailure(call: Call<List<Story>>, t: Throwable) {
                    Log.e(TAG, "api getData: fail")
                }
            })
    }
}
